### DISCUSSION TITLE
PoDynamicFormField - atributo 'params'

### DISCUSSION THREAD

**Discussion by wsteixeira** [po-ui/po-angular#1124] [Tire suas dúvidas]
É possível definir o valor do atributo 'params' de forma dinâmica? Eu gostaria de passar como paramento outro campo do formulário, mas no 'load' os valores do campos ainda não estão carregados. Eu tentei o ngDoCheck, o atributo 'params' foi alterado, mas não foi usado na requisição do 'searchService' o valor original definido no 'PoDynamicFormField':

```
  ngDoCheck() {
    if (this.dynamicForm.model.vendorCode && !this.formloaded) {
      console.log('ngDoCheck :', this.dynamicForm.model);
      this.formloaded = true;
      this.dynamicForm.fields.find(field => {
        return field.property === 'documentId';
      }).params = { vendorCode: this.dynamicForm.model.vendorCode };
      console.log('ngDoCheck :', this.dynamicForm);
    }
  }


```

**Comment by alinelariguet** [comment]
A propriedade p-load pode ser usada com uma string (endpoint), ou pode ser usada passando uma função, nessa função você pode fazer a requisição e passar os params como desejar, veja se isso funciona para você.
Atenciosamente

**Reply by wsteixeira** [reply]
Boa tarde,

Aline, obrigado pelo retorno, mas vou explicar melhor a situação.

Eu estou falando do atributo "params" do "PoPageDynamicEditField".

Vamos imaginar uma interface do "PoPageDynamicEditComponent" onde eu tenha os campos "state" e "city" onde eu vou usar o atributo "searchService" no campo "city", assim o lookup terá mais de 5 mil cidades e para filtrar seria interessante usar o atributo "params".

- Inclusão, eu posso usar o atributo "validade" no campo "state"  que dispara um evento de mudança, onde eu posso atribuir um novo valor ao atributo "params" como por exemplo "params: { state: 'SP' }" do campo "city".
- Alteração, caso eu queira alterar diretamente a cidade, eu já teria que atribuir o valor do "params" logo após a carga dos dados, pois antes disso não tenho a informação do estado para atribuir no "params". Eu até conseguir atribuir o valor conforme demonstrei no exemplo que passei, mas o problema é que internamente o componente não usa o valor que eu passei, ou seja, deve estar usando uma informação atribuída anteriormente em outra variável, objeto e etc.
